### ISSUE TITLE
ImageNewCmap could return SDL_Surface with null pixel pointer

### ISSUE THREAD

**Issue by publicqi** [libsdl-org/SDL_image#180]
```c
    image = ImageNewCmap(len, height, cmapSize);

    for (i = 0; i < cmapSize; i++)
        ImageSetCmap(image, i, cmap[CM_RED][i],
                     cmap[CM_GREEN][i], cmap[CM_BLUE][i]);

    while ((v = LWZReadByte(src, FALSE, c, state)) >= 0) {
        ((Uint8 *)image->pixels)[xpos + ypos * image->pitch] = (Uint8)v;
```
In IMG_gif.c:644, ImageNewCmap may return a SDL_Surface with null pixel pointer if either width or height is 0. Afterwards, image->pixel is dereferenced.

**Comment by sezero** [comment]
Fixed by https://github.com/libsdl-org/SDL_image/commit/6f6dd3c4ad2d46936d014b977a274071b5cfa5ae in main, and by https://github.com/libsdl-org/SDL_image/commit/149794973d24ec38d3b62e46c0b8c3484f2b115b in SDL-1.2 branches.

**Comment by publicqi** [comment]
It's not returning NULL but returning a SDL_Surface with NULL pixel. The actual crash will happen in 
```
((Uint8 *)image->pixels)[xpos + ypos * image->pitch] = (Uint8)v;
```

**Comment by publicqi** [comment]
`image` can also be NULL. The check is correct but additional condition should be added to check if `image->pixels` is NULL.

**Comment by sezero** [comment]
Can you give me a file that triggers this?

**Comment by publicqi** [comment]
python -c "print('GIF89a0000000,000000\x00\x000\x08')"

**Comment by sezero** [comment]
Fixed in main/2.0 with https://github.com/libsdl-org/SDL_image/commit/5f73c9e7d0f84b4b574201ecba01bd0d1ef48669 and in SDL-1.2 with https://github.com/libsdl-org/SDL_image/commit/bd37d9aa579e7616712b0a0baf2aceab2c5833cb

Thanks.
